receiver/mongodbreceiver/client.go

```diff
@@ -79,6 +83,38 @@ func (c *mongodbClient) DBStats(ctx context.Context, database string) (bson.M, e
 	return c.RunCommand(ctx, database, bson.M{"dbStats": 1})
 }
 
+// TopStats is an admin command that return the result of db.adminCommand({ top: 1 })
+// more information can be found here: https://www.mongodb.com/docs/manual/reference/command/top/
+func (c *mongodbClient) TopStats(ctx context.Context) (bson.M, error) {
+	return c.RunCommand(ctx, "admin", bson.M{"top": 1})
+}
+
+// ListCollectionNames returns a list of collection names for a given database
+// SetAuthorizedCollections allows a user without the required privilege to run the command ListCollections.
+// more information can be found here: https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.9.0/mongo#Database.ListCollectionNames
+func (c *mongodbClient) ListCollectionNames(ctx context.Context, database string) ([]string, error) {
+	lcOpts := options.ListCollections().SetAuthorizedCollections(true)
```

Comment:
This option will allow users with the clusterMonitor role to access the list collection names without having to add a read role for each database.

Reply:
How would a user change this option?

Reply:
Right now A user can't, meaning that for every collection/database pair an `index.access.count` metric will be collected. An individual can choose to focus on certain index.access metrics that are being collected and ignore the rest if it is of no interest to the individual collecting the metrics. This is by far the simplest approach.

Without this option, the alternative would be to run with an admin privileged to get the same effect or...
to have a user with a clusterMonitor role and read role for each database they want this metric recorded for. 

So if a user could change this option it could be through the config.yaml to set `authorize_collections`or some other name  to false. Then if that flag it set and the individual did not add a clusterMontior role and a user read role the same error would appear. If the user did add the appropriate role then no error would appear. One weird effect would be that running as admin would still give the the results for all db/collections even if the `authorize_collections` was set to true or false. If admin and the clusterMonitor role were to give the same output then instead of a bool flag in the config a list of databases to collect from would be needed.

Personally, I think defaulting the user to collect all the index.access.count metrics as is with no config option would be the simplest approach for users and not add on tons of documentation.

Reply:
> running as admin would still give the the results for all db/collections even if the authorize_collections was set to true or false

Just to confirm, there is no way for us to control whether or not we are requesting admin dbs/collections? If so, I agree there's no point in having a config value for this. It would only make sense if we can make requests that correlate to the configuration.

Reply:
Correct

Reply:
From what I am reading, the admin database is always called exactly `admin`. If that's the case, can't we just do:

```go
func (c *mongodbClient) ListCollectionNames(ctx context.Context, database string) ([]string, error) {
	if database == "admin" && !c.collectAdmin {
		return ...
	}
	lcOpts := options.ListCollections().SetAuthorizedCollections(true)
	return c.Database(database).ListCollectionNames(context.Background(), bson.D{}, lcOpts)
}
```

Or better yet, check the database name before before calling the function.

Reply:
I did more testing and found that actually having read permission is not enough for a user role and will still give an error. Below is an example of the output 

Example output format is in `dbName[collection names]`.

- ListCollectionNames as admin returns all collection names.
        - example: `admin:[system.users, system.version], config[system.sessions], local[startup_log] testdb[orders, products]`
- ListCollectionNames without SetAuthorizedCollections will return an error with and without the roles listed as read only.
- ListCollectionNames with SetAuthorizedCollections(true) will return some collection names.
        - example: `admin[] config[] local[startup_log] testdb[orders, products]`

So running as admin will give a few more metrics for admin and config db compared to running with the setAuthorizedCollections.

Reply:
Running as admin will typically result in access to more information, but this is not always possible or advisable for users.

I just looked up what `SetAuthorizedCollections` [says it does](https://github.com/mongodb/mongo-go-driver/blob/a0066218d8942310155b64057c094448e90e3356/mongo/options/listcollectionsoptions.go#L17-L19), and it `limits the documents returned to only contain collections the user is authorized to use`. This sounds like exactly what we want, regardless of user. Can you can confirm that using this option does not yield uncontrolled errors when run as non-admin?

Going back to what I suggested earlier:

> 1. Gracefully handle the case where we cannot collect them. Possibly this means requiring a config flag to enable collection of these metrics.
> 2. Suggest the lesser permissions as LPU.
> 3. Clearly document which additional permissions are necessary to collect additional metrics.

1.  If `SetAuthorizedCollections(true)` is graceful, then I don't think we need a config flag.
2 & 3. These still appear to be the correct actions for us to take.

WDYT @JonathanWamsley?

Reply:
I can confirm that there are no uncontrolled errors when ran on a non-admin.

Yes, I can create a LPU setup script and run an integration test. I will then suggest this lesser permission LPU in the readme and outline that index.access.count will be available at 4.0+.